test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthCompilationTests.cs

```diff
@@ -0,0 +1,126 @@
+                {
+                    foreach (var runtime in TestUtils.GetCoreClrRuntimeComponents())
+                    {
+                        yield return runtime;
+                    }
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(RuntimeComponents))]
+        public void DthCompilation_GetDiagnostics_OnEmptyConsoleApp(string flavor, string os, string architecture)
+        {
+            var projectName = "EmptyConsoleApp";
+            var runtimeHomePath = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+            var testProject = _fixture.GetTestProjectPath(projectName);
+
+            using (var server = DthTestServer.Create(runtimeHomePath, testProject))
+            using (var client = new DthTestClient(server, 0))
+            {
+                // Drain the inital messages
+                client.Initialize(testProject);
+                client.DrainMessages();
+
+                client.SendPayLoad("GetDiagnostics", new { });
+                var message = client.DrainMessages().Single(msg => msg.MessageType == "AllDiagnostics");
+                message.EnsureSource(server, client);
+
+                var diagnosticsGroups = (JArray)message.Payload;
+                Assert.Equal(4, diagnosticsGroups.Count);
+
+                foreach (var df in diagnosticsGroups.OfType<JObject>())
+                {
+                    if (RuntimeEnvironmentHelper.IsMono && df["Framework"]["ShortName"].Value<string>() == "net46")
+                    {
+                        continue;
+                    }
+
+                    var errors = (JArray)df["Errors"];
+                    Assert.False(errors.Any(), $"Unexpected compilation errors {string.Join(", ", errors.Select(e => e.Value<string>()))}");
+
+                    var warnings = (JArray)df["Warnings"];
+                    Assert.False(warnings.Any(), $"Unexpected compilation warnings {string.Join(", ", warnings.Select(e => e.Value<string>()))}");
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(RuntimeComponents))]
+        public void DthCompilation_GetCompiledAssembly_OnEmptyConsoleApp(string flavor, string os, string architecture)
+        {
+            var projectName = "EmptyLibrary";
+            var runtimeHomePath = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+
+            string testProject;
+            using (_fixture.CreateDisposableTestProject(projectName, runtimeHomePath, out testProject))
+            using (var server = DthTestServer.Create(runtimeHomePath, testProject))
+            using (var client = new DthTestClient(server, 0))
+            {
+                JToken value;
+
+                // Drain the inital messages
+                client.Initialize(testProject);
+
+                var before = client.DrainMessages().Single(msg => msg.MessageType == "Dependencies").Payload["Dependencies"] as JObject;
```

Comment:
Can we make a more first class way of getting specific message types out of the client API? This looks fragile.